src/Sulu/Bundle/EventLogBundle/Infrastructure/Doctrine/Repository/EventRecordRepository.php

```diff
@@ -62,13 +68,51 @@ public function createForDomainEvent(DomainEvent $domainEvent): EventRecordInter
     {
-        $this->entityManager->flush();
+        // use query-builder to insert only given entity instead of flushing all managed entities via the entity-manager
+        // this prevents flushing unrelated changes and allows to call this method in a postFlush event-listener
+        $classMetadata = $this->entityManager->getClassMetadata($this->getClassName());
+
+        $queryBuilder = $this->getQueryBuilder();
+        $queryBuilder
+            ->insert($classMetadata->getTableName())
+            ->setValue($classMetadata->getColumnName('eventType'), $queryBuilder->createNamedParameter($eventRecord->getEventType()))
+            ->setValue($classMetadata->getColumnName('eventContext'), $queryBuilder->createNamedParameter(\json_encode($eventRecord->getEventContext())))
+            ->setValue($classMetadata->getColumnName('eventDateTime'), $queryBuilder->createNamedParameter($eventRecord->getEventDateTime()->format('Y-m-d H:i:s')))
+            ->setValue($classMetadata->getColumnName('eventBatch'), $queryBuilder->createNamedParameter($eventRecord->getEventBatch()))
+            ->setValue($classMetadata->getColumnName('resourceKey'), $queryBuilder->createNamedParameter($eventRecord->getResourceKey()))
+            ->setValue($classMetadata->getColumnName('resourceId'), $queryBuilder->createNamedParameter($eventRecord->getResourceId()))
+            ->setValue($classMetadata->getColumnName('resourceLocale'), $queryBuilder->createNamedParameter($eventRecord->getResourceLocale()))
+            ->setValue($classMetadata->getColumnName('resourceWebspaceKey'), $queryBuilder->createNamedParameter($eventRecord->getResourceWebspaceKey()))
+            ->setValue($classMetadata->getColumnName('resourceTitle'), $queryBuilder->createNamedParameter($eventRecord->getResourceTitle()))
+            ->setValue($classMetadata->getColumnName('resourceTitleLocale'), $queryBuilder->createNamedParameter($eventRecord->getResourceTitleLocale()))
+            ->setValue($classMetadata->getColumnName('resourceSecurityContext'), $queryBuilder->createNamedParameter($eventRecord->getResourceSecurityContext()))
+            ->setValue($classMetadata->getColumnName('resourceSecurityObjectType'), $queryBuilder->createNamedParameter($eventRecord->getResourceSecurityObjectType()))
+            ->setValue($classMetadata->getColumnName('resourceSecurityObjectId'), $queryBuilder->createNamedParameter($eventRecord->getResourceSecurityObjectId()));
+
+        if (null !== $eventRecord->getUser()) {
+            $queryBuilder->setValue($classMetadata->getColumnName('userId'), $queryBuilder->createNamedParameter($eventRecord->getUser()->getId()));
+        }
+
+        if ($this->shouldPersistPayload) {
+            $queryBuilder->setValue('eventPayload', $queryBuilder->createNamedParameter(\json_encode($eventRecord->getEventPayload())));
+        }
+
+        if (!$classMetadata->idGenerator->isPostInsertGenerator()) {
```

Comment:
maybe add a comment that tries to explain this

```suggestion
        // set value of id explicitly if class has a pre-insert identity-generator to be compatible with postgresql
        if (!$classMetadata->idGenerator->isPostInsertGenerator()) {
```